doc/source/datainputs.rst

```diff
@@ -16,28 +16,128 @@ To simplify the examples below, we will assume in all cases that the user wishes
+Here, we provide an example that reads data from a POS file.
+
+.. code-block:: python
+
+    import garnett
+
+    with garnett.read('trajectory.pos') as traj:
+        for frame in traj:
+            rdf.compute(system=frame, reset=False)
+
+OVITO Modifiers
+---------------
+
+The `OVITO Open Visualization Tool <https://www.ovito.org/>`__ supports user-written Python modifiers.
+The **freud** package can be installed alongside OVITO to enable user modifiers that leverage analyses from **freud**.
+Below is an example modifier that creates a user particle property in the OVITO pipeline for Steinhardt bond order parameters.
 
+.. code-block:: python
+
+    import freud
+
+    def modify(frame, data):
+        ql = freud.order.Steinhardt(l=6)
+        ql.compute(system=data, neighbors={'num_neighbors': 6})
+        data.create_user_particle_property(
+            name='ql', data_type=float, data=ql.ql)
+        print('Created ql property for {} particles.'.format(data.particles.count))
+
+HOOMD-blue Snapshots
+--------------------
+
+`HOOMD-blue <https://hoomd-blue.readthedocs.io/>`__ supports analyzers, callback functions that can perform analysis.
+Below is an example demonstrating how to use an anlyzer to log the Steinhardt bond order parameter :math:`q_6` during the simulation run.
+
+.. code-block:: python
+
+    import hoomd
+    from hoomd import md
+    import freud
 
-Reading Text Files (XYZ files)
-==============================
+    hoomd.context.initialize()
 
-Text files, such as those in the XYZ format, are common simulation data outputs.
+    # Create a 10x10x10 simple cubic lattice of particles with type name A
+    system = hoomd.init.create_lattice(
+        unitcell=hoomd.lattice.sc(a=2.0, type_name='A'), n=10)
+
+    # Specify Lennard-Jones interactions between particle pairs
+    nl = md.nlist.cell()
+    lj = md.pair.lj(r_cut=3.0, nlist=nl)
+    lj.pair_coeff.set('A', 'A', epsilon=1.0, sigma=1.0)
+
+    # Integrate at constant temperature
+    md.integrate.mode_standard(dt=0.005)
+    hoomd.md.integrate.langevin(group=hoomd.group.all(), kT=1.2, seed=4)
+
+    # Create a Steinhardt object to analyze snapshots
+    ql = freud.order.Steinhardt(l=6)
+
+    def compute_q6(timestep):
+        print(timestep)
+        snap = system.take_snapshot()
+        ql.compute(system=snap, neighbors={'num_neighbors': 6})
+        return ql.order
+
+    # Register a logger that computes q6 and saves to a file
+    ql_logger = hoomd.analyze.log(filename='ql.dat', quantities=['q6'], period=100)
+    ql_logger.register_callback('q6', compute_q6)
+
+    # Run for 10,000 time steps
+    hoomd.run(10e3)
+
+Reading Text Files
+==================
+
+Typically, it is best to use one of the natively supported data readers described above, however it is sometimes necessary to parse trajectory information directly from a text file.
 XYZ files can be generated and used by many tools for particle simulation and analysis, including LAMMPS and VMD.
```

Comment:
One example of a plain text format is the XYZ file format, which can be generated and used by many tools for particle simulation and analysis, including LAMMPS and VMD.